From 40123757a799fac0446e33055b33a540bf725589 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Fri, 7 Jun 2019 23:17:13 +0200
Subject: [PATCH] meson.build: fix static build with gcrypt

gcrypt depends on gpg-error so save the result of
dependency('gpg-error') in gpg_error_dep and use it when needed to fix
static build

Fixes:
 - http://autobuild.buildroot.net/results/fb698e3e903869978bd5e69d791ec362317b7981

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Upstream status: https://github.com/iputils/iputils/pull/185]
---
 meson.build        | 5 ++++-
 ninfod/meson.build | 2 +-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 84e4360..24bda0c 100644
--- a/meson.build
+++ b/meson.build
@@ -119,6 +119,9 @@ elif crypto == 'none'
 		description : 'If set RFC6744 random does not use any CRYPTO lib.')
 endif
 
+# gpg-error is a dependency of gcrypt
+gpg_error_dep = dependency('gpg-error', required : false)
+
 systemd = dependency('systemd', required : false)
 systemdunitdir = get_option('systemdunitdir')
 if systemdunitdir == '' and systemd.found()
@@ -241,7 +244,7 @@ libcommon = static_library(
 
 if build_ping == true
 	executable('ping', ['ping.c', 'ping_common.c', 'ping6_common.c', git_version_h],
-		dependencies : [m_dep, cap_dep, idn_dep, crypto_dep, resolv_dep],
+		dependencies : [m_dep, cap_dep, idn_dep, crypto_dep, gpg_error_dep, resolv_dep],
 		link_with : [libcommon],
 		install: true)
 	meson.add_install_script('build-aux/setcap-setuid.sh',
diff --git a/ninfod/meson.build b/ninfod/meson.build
index a0ee94d..1e7aa26 100644
--- a/ninfod/meson.build
+++ b/ninfod/meson.build
@@ -10,7 +10,7 @@ ninfod_sources = files('''
 	ninfod_name.c
 '''.split())
 executable('ninfod', [ninfod_sources, git_version_h],
-	dependencies : [cap_dep, crypto_dep, rt_dep, threads],
+	dependencies : [cap_dep, crypto_dep, gpg_error_dep, rt_dep, threads],
 	link_with : [libcommon],
 	include_directories : inc,
 	install: true,
-- 
2.20.1

